src/colorizer/Dataset.ts

```diff
@@ -362,17 +407,53 @@ export default class Dataset {
       return undefined;
     }
 
-    // Allow for undefined or null backdrop frames in the manifest
-    if (this.frameFiles[index] === undefined || this.frameFiles[index] === null) {
-      return undefined;
-    }
-
-    const fullUrl = this.resolveUrl(frames[index]);
+    const fullUrl = this.resolvePathToUrl(frames[index]);
     const loadedFrame = await this.backdropLoader.load(fullUrl);
     this.backdropFrames?.insert(cacheKey, loadedFrame);
     return loadedFrame;
   }
 
+  /**
+   * Returns a lookup table of ID offsets for each time in the dataset, used to
+   * get the global indices for objects in the image/frame data.
+   *
+   * When looking up data for an object with segmentation ID `segId` in the
+   * image/frame data at time `t`, the global index of the object is given by
+   * `segId + frameToIdOffsets[t]`.
+   *
+   * NOTE: This makes a lot of VERY LARGE assumptions about the frame data,
+   * namely that either:
+   * - Frame data has segmentation IDs that are globally unique, OR
+   * - Frame data has segmentation IDs that increase monotonically (e.g. 2, 3,
+   *   4, 5, etc.) on that frame.
+   *
+   * The second assumption breaks down if there are any skipped segmentation IDs
+   * in the frame. A future, more robust version of this is described in
+   * https://github.com/allen-cell-animated/timelapse-colorizer/issues/630.
+   */
```

Comment:
This ends up getting replaced in the next PR, pt. 5 (support arbitrary segmentation IDs)